test/integration_test/common_test.go

```diff
@@ -3,27 +3,75 @@
 package integrationtest
 
 import (
+	"flag"
 	"os"
 	"path"
+	"strings"
 	"testing"
+	"time"
 
 	corev1 "github.com/libopenstorage/operator/pkg/apis/core/v1"
 	k8sutil "github.com/libopenstorage/operator/pkg/util/k8s"
+	testutil "github.com/libopenstorage/operator/pkg/util/test"
 	"github.com/portworx/sched-ops/k8s/operator"
+	v1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 )
 
 const (
 	specDir = "./operator-test"
 
-	defaultPxSpecGenURL      = "http://install.portworx.com"
-	defaultPxSpecGenEndpoint = ""
+	defaultPxNamespace              = "kube-system"
+	pxReleaseManifestURLEnvVariable = "PX_RELEASE_MANIFEST_URL"
+
+	defaultValidateStorageClusterTimeout       = 900 * time.Second
+	defaultValidateStorageClusterRetryInterval = 30 * time.Second
+	defaultValidateUninstallTimeout            = 900 * time.Second
+	defaultValidateUninstallRetryInterval      = 30 * time.Second
 )
 
+var pxSpecGenURL string
+var pxEndpoint string
+
 func TestMain(m *testing.M) {
+	flag.StringVar(&pxSpecGenURL,
+		"portworx-spec-gen-url",
+		"",
+		"Portworx Spec Generator URL")
+	flag.StringVar(&pxEndpoint,
+		"portworx-endpoint",
+		"",
+		"Portworx Endpoint that defines the version of Portworx and its components")
+	flag.Parse()
 	os.Exit(m.Run())
 }
 
+// Here we make StorageCluster object and add all the common basic parameters that all StorageCluster should have
+func constructStorageCluster() (*corev1.StorageCluster, error) {
```

Comment:
If we want to test a released version of Portworx, how would this work? 
I assume, we would only pass the endpoint in that case. But here we are not setting the portworx version, so the operator will end up installing the latest available version every time, right?

Reply:
Added setting imageName for all deployments as well as EnvVar in case of `edge`.